src/YesSql.Abstractions/TypeService.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+
+namespace YesSql
+{
+    public class TypeService : ITypeService
+    {
+        private readonly ConcurrentDictionary<Type, string> typeNames = new ConcurrentDictionary<Type, string>();
+
+        public string this[Type t]
+        {
+            get
+            {
+                var typeInfo = t.GetTypeInfo();
```

Comment:
This and the `customName` generation should be in the concurrentdictionary lambda